stateMutators.js

```diff
@@ -1,205 +1,197 @@
     return {
         ...state,
         basket: {
-            ...basket,
+            ...state.basket,
             basketValue: value,
         },
     }
-};
+}
 
-const resetBasketValue = (state) => {
+export const resetBasketValue = (state) => {
     return {
         ...state,
         basket: {
-            ...basket,
-            basketValue: 0,
+            ...state.basket,
+            basketValue: INIT_STATE.basket.basketValue,
         },
     }
-};
+}
 
-const setScore = (state, value) => {
+export const setScore = (state, value) => {
     return {
         ...state,
         score: value,
     }
-};
+}
 
-const resetScore = (state) => {
+export const resetScore = (state) => {
     return {
         ...state,
-        score: 0,
+        score: INIT_STATE.score,
     }
-};
+}
 
-const resetLivesRemaining = (state) => {
+export const resetLivesRemaining = (state) => {
     return {
         ...state,
-        livesRemaining: 3,
+        livesRemaining: INIT_STATE.livesRemaining,
     }
-};
+}
 
-const setGameLevel = (state, value) => {
+export const setGameLevel = (state, value) => {
     return {
         ...state,
         gameLevel: value,
     }
-};
+}
 
-const resetGameLevel = (state) => {
+export const resetGameLevel = (state) => {
     return {
         ...state,
-        gameLevel: 0,
+        gameLevel: INIT_STATE.gameLevel,
     }
-};
+}
 
-const setGameMode = (state, value) => {
+export const setGameMode = (state, value) => {
     return {
         ...state,
         gameMode: value,
     }
-};
+}
 
-const addFallingObject = (state, fallingObj) => {
+export const addFallingObject = (state, fallingObj) => {
     return {
         ...state,
-        fallingObjects: [
-            ...state.fallingObjects,
-            fallingObj,
-        ]
+        fallingObjects: [...state.fallingObjects, fallingObj],
+    }
+}
+
+export const removeFallingObject = (state, fallingObjId) => {
+    const index = state.fallingObjects.findIndex((obj) => obj.id === id)
+    if (index === -1) {
+        return state // A non-existent ID was passed in, hence do nothing
     }
-};
 
-const removeFallingObject = (state, fallingObjId) => {
     return {
         ...state,
         fallingObjects: [
-            ...state.fallingObjects.slice(0, fallingObjId),
-            ...state.fallingObjects.slice(fallingObjId + 1),
-        ]
+            ...state.fallingObjects.slice(0, index),
+            ...state.fallingObjects.slice(index + 1),
+        ],
     }
-};
-
+}
 
 // Object constructor for falling object
-function FallingObject(posX, posY, numerator, denominator) {
-    this.xPos = xPos;
-    this.yPos = yPos;
-    this.width = width;
-    this.height = height;
-    this.numerator = numerator;
-    this.denominator = denominator;
-    this.value = numerator / denominator;
-    this.id = id;
+export function FallingObject({
+    posX,
+    posY,
+    numerator,
+    denominator,
+    id,
+    width = FALLING_OBJ_SIZE,
```

Comment:
how come both the width and height have the same FALLING_OBJ_SIZE? Is it because we always want it to be a square?

Reply:
Yeah that's what I was thinking. This is just set as the default values though, i.e., if you don't explictly set `width` and `height` when creating a `FallingObject`

Reminds me I made a few fixes to the mutators code like this in 0e4745a8395e05c27e5aec7181086bed8911aa15 , if you haven't already seen it. Some comments in that commit message

Reply:
Thank you for that. So many rookie errors from the rookie